app/src/main/java/com/zulip/android/networking/AsyncLogin.java

```diff
@@ -16,23 +19,28 @@
     public static final String UNREGISTERED = "unregistered";
     public static final String DISABLED = "disabled";
 
-    LoginActivity context;
+    Context context;
     boolean userDefinitelyInvalid = false;
+    boolean devServer = true; //If this is a DevAuthBackend server!
 
-    public AsyncLogin(LoginActivity loginActivity, String username, String password) {
-        super((ZulipApp) loginActivity.getApplication());
-        context = loginActivity;
+    public AsyncLogin(Context context, String username, String password, boolean devServer) {
```

Comment:
Is there a reason that this isn't of type `Activity`?  Below you're casting it to `Activity` without checking.